File: DataStructure And Algorithms/Practical_3.py
stack = []

# Function to push an element onto the stack
def push(element):
    stack.append(element)
    print("Element pushed onto the stack.")

# Function to pop an element from the stack
def pop():
    if len(stack) > 0:
        stack.pop()
        print("Element popped from the stack.")
    else:
        print("Stack is empty.")

# Function to peek at the top element of the stack
def peep():
    if len(stack) > 0:
        print("Top element of the stack:", stack[-1])
    else:
        print("Stack is empty.")

# Function to change an element at a given index of the stack
def change(index, element):
    if index < len(stack):
        stack[index] = element
        print("Element at index", index, "is changed to", element)
    else:
        print("Invalid index.")

# Function to display the elements of the stack
def display():
    if len(stack) > 0:
        print("Elements of the stack:", stack)
    else:
        print("Stack is empty.")

# Main function to take user choice of operation
def main():
    while True:
        print("\nSelect operation:")
        print("1. Push")
        print("2. Pop")
        print("3. Peep")
        print("4. Change")
        print("5. Display")
        print("6. Exit")
        choice = int(input("Enter choice: "))

        if choice == 1:
            element = input("Enter element to push: ")
            push(element)
        elif choice == 2:
            pop()
        elif choice == 3:
            peep()
        elif choice == 4:
            index = int(input("Enter index to change: "))
            element = input("Enter new element: ")
            change(index, element)
        elif choice == 5:
            display()
        elif choice == 6:
            break
        else:
            print("Invalid choice.")

# Call the main function
if __name__ == '__main__':
    main()

File: DataStructure And Algorithms/Practical_3.java
import java.util.Scanner;

class Stack {
    int top = -1;
    int MAX = 10;
    int[] stack = new int[MAX];

    void push(int value) {
        if (top == MAX - 1) {
            System.out.println("Stack Overflow");
        } else {
            top++;
            stack[top] = value;
            System.out.println("Inserted value: " + value);
        }
    }

    void pop() {
        if (top == -1) {
            System.out.println("Stack Underflow");
        } else {
            System.out.println("Popped value: " + stack[top]);
            top--;
        }
    }

    void peep() {
        if (top == -1) {
            System.out.println("Stack is empty");
        } else {
            System.out.println("Top value: " + stack[top]);
        }
    }

    void change(int index, int value) {
        if (top - index + 1 < 0) {
            System.out.println("Invalid index");
        } else {
            stack[top - index + 1] = value;
            System.out.println("Value changed at index " + index + " to " + value);
        }
    }

    void display() {
        if (top == -1) {
            System.out.println("Stack is empty");
        } else {
            System.out.println("Stack values:");
            for (int i = top; i >= 0; i--) {
                System.out.println(stack[i]);
            }
        }
    }
}

public class Practical_3 {
    public static void main(String[] args) {
        Stack stack = new Stack();
        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.println("\nSelect operation:");
            System.out.println("1. Push");
            System.out.println("2. Pop");
            System.out.println("3. Peep");
            System.out.println("4. Change");
            System.out.println("5. Display");
            System.out.println("6. Exit");
            int choice = sc.nextInt();
            switch (choice) {
                case 1:
                    System.out.print("Enter value: ");
                    int value = sc.nextInt();
                    stack.push(value);
                    break;
                case 2:
                    stack.pop();
                    break;
                case 3:
                    stack.peep();
                    break;
                case 4:
                    System.out.print("Enter index: ");
                    int index = sc.nextInt();
                    System.out.print("Enter value: ");
                    value = sc.nextInt();
                    stack.change(index, value);
                    break;
                case 5:
                    stack.display();
                    break;
                case 6:
                    sc.close();
                    System.exit(0);
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
    
}
